src/Fixer/LanguageConstruct/SingleSpaceAroundConstructFixer.php

```diff
@@ -0,0 +1,462 @@
+        'elseif' => T_ELSEIF,
+        'enum' => null,
+        'extends' => T_EXTENDS,
+        'final' => T_FINAL,
+        'finally' => T_FINALLY,
+        'for' => T_FOR,
+        'foreach' => T_FOREACH,
+        'function' => T_FUNCTION,
+        'function_import' => CT::T_FUNCTION_IMPORT,
+        'global' => T_GLOBAL,
+        'goto' => T_GOTO,
+        'if' => T_IF,
+        'implements' => T_IMPLEMENTS,
+        'include' => T_INCLUDE,
+        'include_once' => T_INCLUDE_ONCE,
+        'instanceof' => T_INSTANCEOF,
+        'insteadof' => T_INSTEADOF,
+        'interface' => T_INTERFACE,
+        'match' => null,
+        'named_argument' => CT::T_NAMED_ARGUMENT_COLON,
+        'namespace' => T_NAMESPACE,
+        'new' => T_NEW,
+        'open_tag_with_echo' => T_OPEN_TAG_WITH_ECHO,
+        'php_doc' => T_DOC_COMMENT,
+        'php_open' => T_OPEN_TAG,
+        'print' => T_PRINT,
+        'private' => T_PRIVATE,
+        'protected' => T_PROTECTED,
+        'public' => T_PUBLIC,
+        'readonly' => null,
+        'require' => T_REQUIRE,
+        'require_once' => T_REQUIRE_ONCE,
+        'return' => T_RETURN,
+        'static' => T_STATIC,
+        'switch' => T_SWITCH,
+        'throw' => T_THROW,
+        'trait' => T_TRAIT,
+        'try' => T_TRY,
+        'type_colon' => CT::T_TYPE_COLON,
+        'use' => T_USE,
+        'use_lambda' => CT::T_USE_LAMBDA,
+        'use_trait' => CT::T_USE_TRAIT,
+        'var' => T_VAR,
+        'while' => T_WHILE,
+        'yield' => T_YIELD,
+        'yield_from' => T_YIELD_FROM,
+    ];
+
+    /**
+     * @var array<string, int>
+     */
+    private array $fixTokenMapFollowedByASingleSpace = [];
+
+    /**
+     * @var array<string, int>
+     */
+    private array $fixTokenMapContainASingleSpace = [];
+
+    /**
+     * @var array<string, int>
+     */
+    private array $fixTokenMapPrecededByASingleSpace = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        if (\defined('T_MATCH')) { // @TODO: drop condition when PHP 8.0+ is required
+            self::$tokenMapFollowedByASingleSpace['match'] = T_MATCH;
+        }
+
+        if (\defined('T_READONLY')) { // @TODO: drop condition when PHP 8.1+ is required
+            self::$tokenMapFollowedByASingleSpace['readonly'] = T_READONLY;
+        }
+
+        if (\defined('T_ENUM')) { // @TODO: drop condition when PHP 8.1+ is required
+            self::$tokenMapFollowedByASingleSpace['enum'] = T_ENUM;
+        }
+
+        $this->fixTokenMapContainASingleSpace = [];
+
+        foreach ($this->configuration['constructs_contain_a_single_space'] as $key) {
+            if (null !== self::$tokenMapContainASingleSpace[$key]) {
+                $this->fixTokenMapContainASingleSpace[$key] = self::$tokenMapContainASingleSpace[$key];
+            }
+        }
+
+        $this->fixTokenMapPrecededByASingleSpace = [];
```

Comment:
Property has `[]` as default value, I believe it's not needed to assign it here and we can just iterate over configuration array?

Same for other properties below.

If it's just for safety (calling `configure()` multiple times), then 👍.

Reply:
yes. it's part of public interface and one can call it any moment. we do it like this for any rule.